core/connection/registry.go

```diff
@@ -17,29 +17,32 @@
 
 package connection
 
+// Factory represents a connection constructor
+type Factory func(stateChannel StateChannel, statisticsChannel StatisticsChannel) (Connection, error)
+
 // Registry holds of all plugable connections
 type Registry struct {
-	creators map[string]Creator
+	creators map[string]Factory
```

Comment:
What's wrong with Creator. func type called factory is a bit confusing. Maybe we can introduce interface with single method Create and call it Factory?

Reply:
done